libs/sdk-backend-tiger/src/backend/tigerSpecificFunctions.ts

```diff
@@ -1050,4 +1067,86 @@ export const buildTigerSpecificFunctions = (
             });
         });
     },
+    getAllCSPDirectives: async (): Promise<Array<ICSPDirective>> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getAllEntitiesCspDirectives({});
+                return result.data?.data || [];
+            });
+        } catch (error) {
+            return [];
+        }
+    },
+    getCSPDirective: async (directiveId: string): Promise<ICSPDirective> => {
+        try {
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.getEntityCspDirectives({ id: directiveId });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            throw convertApiError(error);
+        }
+    },
+    createCSPDirective: async (requestData: ICSPDirective): Promise<ICSPDirective> => {
+        try {
+            const jsonApiCspDirectiveInDocument: JsonApiCspDirectiveInDocument = {
+                data: {
+                    id: requestData.id,
+                    type: JsonApiCspDirectiveInTypeEnum.CSP_DIRECTIVE,
+                    attributes: requestData.attributes,
+                },
+            };
+
+            return await authApiCall(async (sdk) => {
+                const result = await sdk.entities.createEntityCspDirectives({
+                    jsonApiCspDirectiveInDocument,
+                });
+                return result.data?.data;
+            });
+        } catch (error: any) {
+            if (error.response?.status === 400) {
+                const message = error?.response?.data?.detail
+                    ? error?.response?.data?.detail
+                    : "Server error";
+                throw new UnexpectedError(message, error);
+            }
+            throw convertApiError(error);
```

Comment:
Can we delete lines 1107 ~ 1112 and keep just this line only? Logic from lines 1107 ~ 1112 is probably already contained in `convertApiError`...

Reply:
`convertApiError` did not handle the statusCode = 400 case.

Reply:
<img width="691" alt="Screen Shot 2022-10-19 at 16 11 52" src="https://user-images.githubusercontent.com/5329091/196649080-5eedb581-cfbb-4931-9327-7359f3d82812.png">
So we need to handle  specific bad request case to get detailed message

Reply:
ok then